uwblueprint/focus-on-nature
# Session Selection

## Notion ticket link

[Registration Form: Session Selection Page](https://www.notion.so/uwblueprintexecs/Registration-Form-Session-Selection-Page-5a0b924714484094a91ac8ca9a82c521)


## Implementation description
* index.tsx maintains the state of which camp sessions have been selected, and how many have been selected. It uses this data to change the UI.
* CampSessionBoard.tsx is responsible for rendering camp cards for both available (AvailableCampCard.tsx) and waitlist (WaitlistCampCard) camps
* Custom clickable boxes have been used to create camp session cards and achieve the functionality of a form
* Grid/GridItem have been used to achieve responsiveness


## Steps to test
1. Go to http://localhost:3000/register/63139c7bc3d7b55b44a01531
2. Select available camps, check if waitlist camps are disabled and CTA is clickable and is the correct name/design
3. Uncheck all available camps, select some waitlist camps, ensure available camps are disabled and CTA is clickable and is correct name/design
4. Once CTA is clicked, ID's of selected camps will be logged to the console (this data will most likely be used for further actions in signing up a camper)
5. Ensure the page is responsive on desktop, tablet, and phone screen sizes
6. NOTE: Current camp image being shown is static as there seems to be an error rendering image retrieved from the API call

_Demo Videos Below (too big to upload, hence the zip file)_

Functionality:
[Screen Recording 2022-12-23 at 3.36.36 AM.mov.zip](https://github.com/uwblueprint/focus-on-nature/files/10293656/Screen.Recording.2022-12-23.at.3.36.36.AM.mov.zip)

Responsive:
[Screen Recording 2022-12-23 at 3.09.49 AM.mov.zip](https://github.com/uwblueprint/focus-on-nature/files/10293663/Screen.Recording.2022-12-23.at.3.09.49.AM.mov.zip)



## What should reviewers focus on?
* Component organization and overall state management
* Styling/Responsiveness (use of breakpoint-based styling and grid layout)

## Checklist
